include/highfive/bits/H5DataType_misc.hpp

```diff
@@ -125,7 +128,20 @@ inline AtomicType<unsigned long long>::AtomicType() {
     _hid = H5Tcopy(H5T_NATIVE_ULLONG);
 }
 
-// float, double and long double mapping
+// half-float, float, double and long double mapping
+#ifdef H5_USE_HALF_FLOAT
+using float16_t = half_float::half;
+
+template <>
+inline AtomicType<float16_t>::AtomicType()
+{
+    _hid = H5Tcopy(H5T_NATIVE_FLOAT);
+    H5Tset_fields(_hid, 15, 10, 5, 0, 10);
```

Comment:
Can you add a comment here to be explicit?

Reply:
Added comments and fixed clang-format violations. Thanks!